### PR TITLE
Be 82 confirm trainee email

### PR DESCRIPTION
**Pull request by nidaffs**
Create service and controller for trainee email confirm,  in approveUrl include jwt token with trainee information. after succes aprove trainee status - changed. ApproveUrl is created by event logic

### CODE REVIEW

File: src/main/java/com/exadel/project/administrator/service/AdministratorService.java [hunk 1]
```diff
@@ -72,10 +72,10 @@ public AdministratorDto getById(Long id) throws EntityNotFoundException {
     }
 
     public AdministratorDto addAdministrator(AdministratorDto administratorDto) throws EntityAlreadyExistsException {
-        if (administratorRepository.findAdministratorByEmail(administratorDto.getEmail()) != null) {
+        if (administratorRepository.findAdministratorByEmail(administratorDto.getEmail()) != null
+                || administratorRepository.findAdministratorByLogin(administratorDto.getLogin()) != null) {
```
Comment (serkushner) [inline src/main/java/com/exadel/project/administrator/service/AdministratorService.java:76]: Let's move all validations in separate spring component AdministratorValidator

File: src/main/java/com/exadel/project/common/event/OnRegistrationCompleteEvent.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import org.springframework.context.ApplicationEvent;
+
+@Getter
+@Setter
+public class OnRegistrationCompleteEvent extends ApplicationEvent {
+
+    private final AdditionalInfo additionalInfo;
+
+    public OnRegistrationCompleteEvent(
+            Object source, AdditionalInfo additionalInfo) {
```
Comment (serkushner) [inline src/main/java/com/exadel/project/common/event/OnRegistrationCompleteEvent.java:16]: format code

File: src/main/java/com/exadel/project/common/event/OnRegistrationCompleteEvent.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+import com.exadel.project.subject.entity.Subject;
+import com.exadel.project.trainee.entity.AdditionalInfo;
+import lombok.Getter;
+import lombok.Setter;
+import org.springframework.context.ApplicationEvent;
+
+@Getter
+@Setter
+public class OnRegistrationCompleteEvent extends ApplicationEvent {
```
Comment (serkushner) [inline src/main/java/com/exadel/project/common/event/OnRegistrationCompleteEvent.java:11]: TraineeRegistrationCompleteEvent

File: src/main/resources/db/changelog/changelog-additional-info-add-email-not-confirm-trainee-status.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+databaseChangeLog:
+  - changeSet:
+      id: 20210505.[BE-82]-add-email-not-confirm
+      author: nidaff.s@gmail.com
+      preConditions:
+        - onFail: MARK_RAN
+        - not:
+          - sqlCheck:
+              expectedResult: "10"
+              sql: SELECT count(DISTINCT status_trainee) FROM additional_info
```
Comment (serkushner) [inline src/main/resources/db/changelog/changelog-additional-info-add-email-not-confirm-trainee-status.yaml:10]: pls fix preConditions

**Review by nazfurdychka — DISMISSED** [review]

File: src/main/resources/db/changelog/changelog-additional-info-add-email-not-confirm-trainee-status.yaml (hunk 4 above)
Comment (nidaffs) [reply src/main/resources/db/changelog/changelog-additional-info-add-email-not-confirm-trainee-status.yaml]: thanks, fixed it

File: src/main/java/com/exadel/project/common/event/OnRegistrationCompleteEvent.java (hunk 3 above)
Comment (nidaffs) [reply src/main/java/com/exadel/project/common/event/OnRegistrationCompleteEvent.java]: thanks, fixed it

File: src/main/java/com/exadel/project/common/event/OnRegistrationCompleteEvent.java (hunk 2 above)
Comment (nidaffs) [reply src/main/java/com/exadel/project/common/event/OnRegistrationCompleteEvent.java]: thanks, fixed it

File: src/main/java/com/exadel/project/administrator/service/AdministratorService.java (hunk 1 above)
Comment (nidaffs) [reply src/main/java/com/exadel/project/administrator/service/AdministratorService.java]: thanks, fixed it

**Review by AlKaliada — DISMISSED** [review]

**Review by AlKaliada — APPROVED** [review]

**Review by ValentinKhorunzhyn — APPROVED** [review]

**Pull request merged**